internal/service/cluster/resource_cluster.go

```diff
@@ -803,13 +810,21 @@ func resourceMongoDBAtlasClusterUpdate(ctx context.Context, d *schema.ResourceDa
 		}
 	}
 
+	replicationSpecs, err := expandReplicationSpecs(d)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf(errorClusterUpdate, clusterName, err))
+	}
 	if d.HasChange("replication_specs") {
-		replicationSpecs, err := expandReplicationSpecs(d)
+		cluster.ReplicationSpecs = replicationSpecs
+	}
+
+	if v, ok := d.GetOk("provider_region_name"); ok {
+		err = validateProviderRegionName(d.Get("cluster_type").(string), v.(string), replicationSpecs)
 		if err != nil {
-			return diag.FromErr(fmt.Errorf(errorClusterUpdate, clusterName, err))
+			if cluster.ProviderSettings != nil {
+				cluster.ProviderSettings.RegionName = ""
```

Comment:
I think this code will be hard to read in the future. Can we perhaps add an explanation of the why that we had in the comment?

Reply:
added